src/database/migrations/20251212250000-complete-tenant-code-migration.js

```diff
@@ -0,0 +1,471 @@
+							type: Sequelize.QueryTypes.SELECT,
+							transaction,
+						}
+					)
+
+					if (!tableExists[0].exists) {
+						console.log(`⚠️  Table ${tableName} does not exist, skipping`)
+						continue
+					}
+
+					// Check if tenant_code column already exists
+					const columnExists = await queryInterface.sequelize.query(
+						`SELECT EXISTS (SELECT FROM information_schema.columns WHERE table_name = :tableName AND column_name = 'tenant_code')`,
+						{
+							replacements: { tableName },
+							type: Sequelize.QueryTypes.SELECT,
+							transaction,
+						}
+					)
+
+					if (!columnExists[0].exists) {
+						await queryInterface.addColumn(
+							tableName,
+							'tenant_code',
+							{
+								type: Sequelize.STRING(255),
+								allowNull: true, // Start as nullable
+							},
+							{ transaction }
+						)
+						console.log(`✅ Added tenant_code to ${tableName}`)
+					} else {
+						console.log(`✅ ${tableName} already has tenant_code column`)
+					}
+				} catch (error) {
+					console.log(`❌ Error processing tenant_code for ${tableName}: ${error.message}`)
+					throw error // Fail fast to trigger rollback
+				}
+			}
+
+			console.log('\n📝 PHASE 2: Adding organization_code columns...')
+			console.log('='.repeat(50))
+
+			// Add organization_code column to specific tables
+			for (const tableName of tablesNeedingOrgCode) {
+				try {
+					// Check if table exists
+					const tableExists = await queryInterface.sequelize.query(
+						`SELECT EXISTS (SELECT FROM information_schema.tables WHERE table_name = :tableName)`,
+						{
+							replacements: { tableName },
+							type: Sequelize.QueryTypes.SELECT,
+							transaction,
+						}
+					)
+
+					if (!tableExists[0].exists) {
+						console.log(`⚠️  Table ${tableName} does not exist, skipping`)
+						continue
+					}
+
+					// Check if organization_code column already exists
+					const columnExists = await queryInterface.sequelize.query(
+						`SELECT EXISTS (SELECT FROM information_schema.columns WHERE table_name = :tableName AND column_name = 'organization_code')`,
+						{
+							replacements: { tableName },
+							type: Sequelize.QueryTypes.SELECT,
+							transaction,
+						}
+					)
+
+					if (!columnExists[0].exists) {
+						await queryInterface.addColumn(
+							tableName,
+							'organization_code',
+							{
+								type: Sequelize.STRING(255),
+								allowNull: true, // Start as nullable
+							},
+							{ transaction }
+						)
+						console.log(`✅ Added organization_code to ${tableName}`)
+					} else {
+						console.log(`✅ ${tableName} already has organization_code column`)
+					}
+				} catch (error) {
+					console.log(`❌ Error processing organization_code for ${tableName}: ${error.message}`)
+					throw error // Fail fast to trigger rollback
+				}
+			}
+
+			console.log('\n📝 PHASE 3: Adding user_name column to user_extensions...')
+			console.log('='.repeat(50))
+
+			// Handle user_extensions table specifically
```

Comment:
Split this logic to a different migration.
- Add username to user_extensions (nullable)
- Backfill username
- Commit